services/issue/issue.go

```diff
@@ -262,45 +262,27 @@ func deleteIssue(ctx context.Context, issue *issues_model.Issue) error {
 	}
 
 	// delete all database data still assigned to this issue
-	if err := issues_model.DeleteInIssue(ctx, issue.ID,
-		&issues_model.ContentHistory{},
-		&issues_model.Comment{},
-		&issues_model.IssueLabel{},
-		&issues_model.IssueDependency{},
-		&issues_model.IssueAssignees{},
-		&issues_model.IssueUser{},
-		&activities_model.Notification{},
-		&issues_model.Reaction{},
-		&issues_model.IssueWatch{},
-		&issues_model.Stopwatch{},
-		&issues_model.TrackedTime{},
-		&project_model.ProjectIssue{},
-		&repo_model.Attachment{},
-		&issues_model.PullRequest{},
+	if err := db.DeleteBeans(ctx,
+		&issues_model.ContentHistory{IssueID: issue.ID},
+		&issues_model.Comment{IssueID: issue.ID},
+		&issues_model.IssueLabel{IssueID: issue.ID},
+		&issues_model.IssueDependency{IssueID: issue.ID},
+		&issues_model.IssueAssignees{IssueID: issue.ID},
+		&issues_model.IssueUser{IssueID: issue.ID},
+		&activities_model.Notification{IssueID: issue.ID},
+		&issues_model.Reaction{IssueID: issue.ID},
+		&issues_model.IssueWatch{IssueID: issue.ID},
+		&issues_model.Stopwatch{IssueID: issue.ID},
+		&issues_model.TrackedTime{IssueID: issue.ID},
+		&project_model.ProjectIssue{IssueID: issue.ID},
+		&repo_model.Attachment{IssueID: issue.ID},
+		&issues_model.PullRequest{IssueID: issue.ID},
+		&issues_model.Comment{RefIssueID: issue.ID},
+		&issues_model.IssueDependency{DependencyID: issue.ID},
+		&issues_model.Comment{DependentIssueID: issue.ID},
```

Comment:
We have foreign keys?

Reply:
That's exactly my question.

Reply:
As far as I know the Gitea Database has no foreign keys

Reply:
No any foreign keys by design to make scale up easier.

Reply:
> No foreign keys

> by design

Generally, that sounds rather like a paradox to me.
Typically you **want** foreign keys to prevent orphaned data or invalid states.

Reply:
I am a lil skeptical to accept this chang due to cascade delete.